src/services/SearchResults.h

```diff
@@ -82,57 +86,60 @@ class SearchResults
     {
     public:
         MemBlock(unsigned int nAddress, unsigned int nSize)
-            : nAddress(nAddress), nSize(nSize)
+            : m_nAddress(nAddress), m_nSize(nSize)
         {
             if (nSize > sizeof(m_vBytes))
                 m_pBytes = new unsigned char[nSize];
         }
 
         MemBlock(const MemBlock& other) noexcept
-            : MemBlock(other.nAddress, other.nSize)
+            : MemBlock(other.m_nAddress, other.m_nSize)
         {
-            if (nSize > sizeof(m_vBytes))
-                memcpy(m_pBytes, other.m_pBytes, nSize);
+            if (m_nSize > sizeof(m_vBytes))
+                memcpy(m_pBytes, other.m_pBytes, other.m_nAddress);
         }
 
         MemBlock(MemBlock&& other) noexcept
         {
-            nAddress = other.nAddress;
-            nSize = other.nSize;
+            m_nAddress = other.m_nAddress;
+            m_nSize = other.m_nSize;
 
-            if (other.nSize > sizeof(m_vBytes))
+            if (other.m_nSize > sizeof(m_vBytes))
             {
                 m_pBytes = other.m_pBytes;
                 other.m_pBytes = nullptr;
-                other.nSize = 0;
+                other.m_nSize = 0;
             }
             else
             {
                 memcpy(m_vBytes, other.m_vBytes, sizeof(m_vBytes));
             }
         }
 
-        ~MemBlock()
+        ~MemBlock() noexcept
         {
-            if (nSize > sizeof(m_vBytes))
-                delete m_pBytes;
+            if (m_nSize > sizeof(m_vBytes))
+            {
+                delete[] m_pBytes;
+                m_pBytes = nullptr;
+            }
         }
 
-        unsigned char* GetBytes() { return (nSize > sizeof(m_vBytes)) ? m_pBytes : &m_vBytes[0]; }
-        const unsigned char* GetBytes() const { return (nSize > sizeof(m_vBytes)) ? m_pBytes : &m_vBytes[0]; }
+        unsigned char* GetBytes() { return (m_nSize > sizeof(m_vBytes)) ? m_pBytes : &m_vBytes[0]; }
+        const unsigned char* GetBytes() const { return (m_nSize > sizeof(m_vBytes)) ? m_pBytes : &m_vBytes[0]; }
 
-        unsigned int GetAddress() const { return nAddress; }
-        unsigned int GetSize() const { return nSize; }
+        unsigned int GetAddress() const { return m_nAddress; }
+        unsigned int GetSize() const { return m_nSize; }
 
     private:
         union // 8 bytes
         {
-            unsigned char m_vBytes[8];
+            unsigned char m_vBytes[8]{};
```

Comment:
Only one initializer was allowed, this array wouldn't get initialized otherwise. The pointer magically got initialized.